src/modules/auth/services/auth.service.ts

```diff
@@ -55,4 +56,32 @@ export class AuthService implements IAuthService {
       throw new HttpException(`${error}`, HttpStatus.INTERNAL_SERVER_ERROR);
     }
   }
+
+  async generateToken(user: User): Promise<IToken> {
+    try {
+      const payload = { email: user.email, id: user.id };
+      return {
+        token: this.jwtService.sign(payload),
+      };
+    } catch (error) {
+      throw new HttpException(`${error}`, HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+  }
+
+  async forgotPassword(email: string): Promise<boolean> {
+    try {
+      const existingUser = await this.userService.findByEmail(email);
+      if (!existingUser) {
+        throw new BadRequestException(`user with email ${email} does not exists`);
+      }
+      const token = await this.generateToken(existingUser);
+      await this.mailService.sendActivationMail(
+        existingUser.email,
+        `${process.env.RESET_PASSWOPRD_LINK}resetpassword/${token.token}`,
```

Comment:
please check Nest documentation what is correct way to use env variables

Reply:
Fixed using env var
const resetPageLink = this.configService.get<string>('RESET_PASSWOPRD_LINK');